### PR TITLE
Added rotate function to UIImage Extensions

### PR DESCRIPTION
**Pull request by ozgurshn**
I used UIGraphicsImageRenderer as suggested Apple.

### CODE REVIEW

File: CoreMLHelpers/UIImage+Extensions.swift [hunk 1]
```diff
@@ -1,45 +1,75 @@
...
-
-  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
-  IN THE SOFTWARE.
-*/
+ Copyright (c) 2017-2019 M.I. Hollemans
```
Comment (hollance) [inline CoreMLHelpers/UIImage+Extensions.swift:2]: Can you leave the original formatting intact?

File: CoreMLHelpers/UIImage+Extensions.swift [hunk 2]
```diff
@@ -1,45 +1,75 @@
...
-    - Parameters:
-      - scale: If this is 1, `newSize` is the size in pixels.
-  */
-  @nonobjc public func resized(to newSize: CGSize, scale: CGFloat = 1) -> UIImage {
-    let format = UIGraphicsImageRendererFormat.default()
-    format.scale = scale
-    let renderer = UIGraphicsImageRenderer(size: newSize, format: format)
-    let image = renderer.image { _ in
-      draw(in: CGRect(origin: .zero, size: newSize))
+    /**
```
Comment (hollance) [inline CoreMLHelpers/UIImage+Extensions.swift:28]: Same here: keep the original spacing

File: CoreMLHelpers/UIImage+Extensions.swift [hunk 3]
```diff
@@ -1,45 +1,75 @@
...
+        return image
+    }
+    
+        /**
+     Rotates the image.
+     
+     - Parameters:
+     - angle: Angle degree (0-360).
+     */
+    @nonobjc public func rotate(angle:CGFloat)->UIImage
```
Comment (hollance) [inline CoreMLHelpers/UIImage+Extensions.swift:50]: Please use the same formatting conventions as the original code, w.r.t spacing etc: `rotate(angle: CGFloat) -> UIImage`

This applies to the rest of this function too.

File: CoreMLHelpers/UIImage+Extensions.swift [hunk 4]
```diff
@@ -1,45 +1,75 @@
...
+        
+        var newSize = CGRect(origin: CGPoint.zero, size: self.size).applying(CGAffineTransform(rotationAngle: radians)).size
+        
+        // Trim off the extremely small float value to prevent core graphics from rounding it up
+        newSize.width = floor(newSize.width)
+        newSize.height = floor(newSize.height)
+        let renderer = UIGraphicsImageRenderer(size:self.size)
+        
+        let image = renderer.image
+        { rendederContext in
```
Comment (hollance) [inline CoreMLHelpers/UIImage+Extensions.swift:62]: This is a typo

File: CoreMLHelpers/UIImage+Extensions.swift (hunk 3 above)
Comment (hollance) [reply CoreMLHelpers/UIImage+Extensions.swift:50]: I'd prefer `rotate(degrees:)` over `angle` so that it's clear the angle is in degrees.

**Comment by ozgurshn** [comment]
Hi, thanks for the feedbacks. I tried to fix formatting with this [merge](https://github.com/hollance/CoreMLHelpers/pull/39/commits/d2fe324cfbe246ba2c7cf82e6351467e34b6fa34)

**Comment by hollance** [comment]
Great! Thanks again for the contribution. 😃

**Comment by ozgurshn** [comment]
I have been using this repo in every Core ML project. It’s my pleasure to contribute :)

**Pull request merged**